contracts/starknet/lib/voting.cairo

```diff
@@ -221,9 +221,13 @@ namespace Voting:
             voting_strategy_params_flat_len, voting_strategy_params_flat
         )
 
+        let (prev_index) = Voting_num_voting_strategies_store.read()
         unchecked_add_voting_strategies(
-            voting_strategies_len, voting_strategies, voting_strategy_params_all, 0
+            voting_strategies_len, voting_strategies, voting_strategy_params_all, prev_index, 0
         )
+        # Incrementing the voting strategies counter by the number of strategies added
+        Voting_num_voting_strategies_store.write(prev_index + voting_strategies_len)
```

Comment:
eeeh maybe I'm a bit picky but I'd rather have a wrapper function that does all this logic rather than having to manage the `Voting_num_voting_strategies_store` outside of `unchecked_add_voting_strategies`.

I think `unchecked_add_voting_strategies` should do this logic, and maybe have a wrapper inside if (something like `unchecked_add_voting_strategies_recurse`) to handle the recursion. Wdyt?

Reply:
yep agree